system/lib/pthread/proxying_stub.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+// Stub implementation of the proxying queue API that does nothing, for use in
+// single-threaded builds.
+
+#include "proxying.h"
+
+em_proxying_queue* em_proxying_queue_create() {
+  return NULL;
+}
```

Comment:
could these all `abort()`?

Reply:
lgtm with that (seems safer to avoid confusion later), or lgtm if there's a good reason not to as well.

Reply:
Where exactly are the references coming from?    Perhaps we can instead use the musl trick of using weak aliases to dummy functions?

Alternately, what happens if we just include the regular version and let it call the stub pthreads function?

Reply:
That would almost work out-of-the-box, but it does call out to `_emscripten_notify_proxying_queue` in library_pthread.js, so we'd at least need to supply a stub for that. But it would still be a user error to use any of the proxying functions except for async proxying to self, which could still work.

Reply:
> Where exactly are the references coming from? Perhaps we can instead use the musl trick of using weak aliases to dummy functions?

The references are coming from various WasmFS backends that get pulled in by `--whole-archive`. If we used weak aliases, how would that differ from the stub solution here. Would it be cleaner?

Reply:
Very little difference I guess.   Using the weak aliases has the downside that it could hide real link failures by failling back to dummy functions if the real library is missing/invalid for some other reason.